### ISSUE TITLE
Observers fire unexpectedly when monitoring objects changes using wildcards

### ISSUE THREAD

**Issue by giovannipiller** [ractivejs/ractive#2800]
## Description:
Found this weird issue with observers. Tried to simplify it as much as possible.

Observer is monitoring some object's keys.
When we change the key `a`, the observer will respond by changing the key `b`.
This will cause Ractive to somehow believe that `a` has been changed once again, before even noticing that `b` has been changed.

It's much easier to understand by watching it happen.

## Versions affected:
0.8.x, edge

## Reproduction:
https://jsfiddle.net/gigasoft/p6q5sgbu/2/ (open the console)

## Code:
```
const r = window.r = new Ractive({
    el: '#main',
    template: '#template',
    data: {
        myData: {
            a: 0,
            b: 1,
        }
    },
    oninit() {
        this.observe('myData.*', (newData, oldData, keypath, letter) => {
            console.log(`${letter} — '${keypath}' changed to ${newData} (prev ${oldData})`)
            if (letter === 'a') {
                // commenting this out will print only one "a" in the console
                this.set('myData.b', 4);
            }
        }, {
            init: false
        });
    },
    onrender() {
        this.set('myData.a', 3);
    }
});
```

**Comment by giovannipiller** [comment]
**Note:** setting `b` within a `setTimeout` won't trigger the issue.

## Example
Convert `this.set('myData.b', 4);`
to `setTimeout(() => (this.set('myData.b', 4)));`

https://jsfiddle.net/gigasoft/p6q5sgbu/3/

**Comment by evs-chris** [comment]
This is now fixed in edge and 0.8-dev and will go out with 0.8.8. Thanks for the report!
